### PR TITLE
Spec v1.0

### PR DESCRIPTION
**Pull request by sbogusev**

### CODE REVIEW

File: source/elements/oneDNN/include/dnnl.hpp [hunk 1]
```diff
@@ -515,25 +470,6 @@ struct stream {
...
     stream();
 
-    /// Constructs a stream for the specified engine and with behavior
-    /// controlled by the specified flags.
-    ///
-    /// @param aengine Engine to create the stream on.
-    /// @param aflags Flags controlling stream behavior.
-    /// @param attr Stream attributes.
-    stream(const engine &aengine, flags aflags = flags::default_flags,
-            const stream_attr &attr = stream_attr());
```
Comment (emfomenk) [inline source/elements/oneDNN/include/dnnl.hpp]: the ctor is still required, but w/o `stream_attr`

File: source/elements/oneDNN/include/dnnl.hpp [hunk 2]
```diff
@@ -1035,15 +966,6 @@ struct memory {
     ///       buffer.
     memory(const desc &md, const engine &aengine, void *handle);
```
Comment (emfomenk) [inline source/elements/oneDNN/include/dnnl.hpp:978]: we should document that underlying buffer type is device usm

in the product we will state something like this: the underlying buffer depends on the runtime (for ocl -- cl_mem, for sycl -- device usm, etc)

File: source/elements/oneDNN/include/dnnl_sycl.hpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+* Copyright 2016-2020 Intel Corporation
```
Comment (emfomenk) [inline source/elements/oneDNN/include/dnnl_sycl.hpp]: 2020 only

Comment (emfomenk) [reply source/elements/oneDNN/include/dnnl_sycl.hpp]: btw, also we might want to check what to do with copyrights

File: source/elements/oneDNN/source/example.cpp [hunk 4]
```diff
@@ -62,30 +68,36 @@ struct eltwise_forward {
 int N, C, H, W;
 
 // User-owned DPC++ objects
-sycl::device dev; // Device
-sycl::context ctx; // Context
-sycl::queue queue; // Queue
+sycl::device dev {sycl::gpu_selector {}}; // Device
+sycl::context ctx {dev}; // Context
+sycl::queue queue {sycl::gpu_selector {}}; // Queue
```
Comment (emfomenk) [inline source/elements/oneDNN/source/example.cpp]: should it be created based on dev?

File: source/elements/oneDNN/source/example.cpp [hunk 5]
```diff
@@ -62,30 +68,36 @@ struct eltwise_forward {
...
-dnnl::memory mem_src {{{N, C, H, W}, dnnl::memory::data_type::f32,
-                              dnnl::memory::format_tag::nhwc},
-        engine, buf_src};
-dnnl::memory mem_dst {{{N, C, H, W}, dnnl::memory::data_type::f32,
-                              dnnl::memory::format_tag::nhwc},
-        engine, buf_dst};
+dnnl::memory mem_src = dnnl::sycl_interop::make_memory(
+        {{N, C, H, W}, dnnl::memory::data_type::f32,
+                dnnl::memory::format_tag::nhwc},
+        engine, dnnl::memory_kind::usm_device, buf_src);
```
Comment (emfomenk) [inline source/elements/oneDNN/source/example.cpp]: memory with usm could be created with regular ctor. I suggest simplify the code here.

File: source/elements/oneDNN/source/replacements.inc.rst [hunk 6]
```diff
@@ -325,9 +325,9 @@
 .. |memory::get_data_handle| replace:: :any:`dnnl::memory::get_data_handle`
 .. |memory::get_desc| replace:: :any:`dnnl::memory::get_desc`
 .. |memory::get_engine| replace:: :any:`dnnl::memory::get_engine`
-.. |memory::get_sycl_buffer| replace:: :any:`dnnl::memory::get_sycl_buffer`
+.. |memory::get_buffer| replace:: :any:`dnnl::memory::get_buffer`
+.. |memory::set_buffer| replace:: :any:`dnnl::memory::set_buffer`
```
Comment (emfomenk) [inline source/elements/oneDNN/source/replacements.inc.rst]: we use set_data_handle in the api. probably should indeed change to `set_buffer()` (see dnnl_sycl.hpp`)

File: source/elements/oneDNN/include/dnnl.hpp (hunk 1 above)
Comment (sbogusev) [reply source/elements/oneDNN/include/dnnl.hpp]: Indeed I have unnecessarily removed it, thanks for noticing.

File: source/elements/oneDNN/source/example.cpp (hunk 4 above)
Comment (sbogusev) [reply source/elements/oneDNN/source/example.cpp]: Indeed we can do it, will update.

File: source/elements/oneDNN/include/dnnl_sycl.hpp (hunk 3 above)
Comment (sbogusev) [reply source/elements/oneDNN/include/dnnl_sycl.hpp]: Updated, thanks for noticing. What do you mean by 

> what to do with copyrights

File: source/elements/oneDNN/source/replacements.inc.rst (hunk 6 above)
Comment (sbogusev) [reply source/elements/oneDNN/source/replacements.inc.rst]: Updated `set_data_handle` to `set_buffer` in `dnnl_sycl.hpp`

File: source/elements/oneDNN/include/dnnl.hpp (hunk 2 above)
Comment (sbogusev) [reply source/elements/oneDNN/include/dnnl.hpp:978]: Included clarifications for `handle` and for the second constructor, that the underlying buffer will be `usm_device` if the user does not provide any memory.

File: source/elements/oneDNN/source/example.cpp (hunk 5 above)
Comment (sbogusev) [reply source/elements/oneDNN/source/example.cpp]: On one hand I would like to keep it to emphasize the flexibility of the API and make as "default" for the user. On the other hand, the simplified constructor indeed may work well as a decent default option for most users.

Will update.

File: source/elements/oneDNN/include/dnnl_sycl.hpp (hunk 3 above)
Comment (emfomenk) [reply source/elements/oneDNN/include/dnnl_sycl.hpp]: ah, never mind (resolved)

**Pull request closed without merging**